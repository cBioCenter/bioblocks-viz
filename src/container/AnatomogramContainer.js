"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
// tslint:disable-next-line:import-name
var anatomogram_1 = require("anatomogram");
var immutable_1 = require("immutable");
var React = require("react");
var react_redux_1 = require("react-redux");
var redux_1 = require("redux");
var action_1 = require("~bioblocks-viz~/action");
var component_1 = require("~bioblocks-viz~/component");
var container_1 = require("~bioblocks-viz~/container");
var data_1 = require("~bioblocks-viz~/data");
var helper_1 = require("~bioblocks-viz~/helper");
var reducer_1 = require("~bioblocks-viz~/reducer");
var selector_1 = require("~bioblocks-viz~/selector");
var AnatomogramContainerClass = /** @class */ (function (_super) {
    tslib_1.__extends(AnatomogramContainerClass, _super);
    function AnatomogramContainerClass(props) {
        var _this = _super.call(this, props) || this;
        _this.divRef = null;
        _this.onClick = function (ids) {
            // Anatomogram returns an array of strings for click events, but we only ever work with a single id.
            var id = ids[0];
            var _a = _this.props, addLabel = _a.onLabelAdd, removeLabel = _a.onLabelRemove, selectIds = _a.selectIds;
            if (selectIds.includes(id)) {
                removeLabel(id);
            }
            else {
                addLabel(id);
            }
        };
        _this.deriveIdsFromSpecies = function (species) { return Object.keys(data_1.AnatomogramMapping[species]); };
        _this.onMouseOut = function (id) {
            return;
        };
        _this.onMouseOver = function (id) {
            return;
        };
        _this.parseCategory = function (category) {
            var splitCategories = category.split(/-|_/);
            return splitCategories[0];
        };
        _this.resizeSVGElement = function (error, svgDomNode) {
            if (_this.divRef) {
                var isSvgHeightBigger = svgDomNode.height.baseVal.value > svgDomNode.width.baseVal.value;
                // The Anatomogram Component internally sets the svg height to 'auto'.
                // So, to allow more flexibility in sizing it, we have to manually override it here. Sorry.
                svgDomNode.style.height = isSvgHeightBigger ? "calc(" + _this.divRef.style.height + " - 50px)" : 'auto';
                svgDomNode.style.padding = '0';
                svgDomNode.style.width = isSvgHeightBigger ? 'auto' : "calc(" + _this.divRef.style.width + " - 75px)";
            }
        };
        _this.state = {
            ids: _this.deriveIdsFromSpecies(props.species),
        };
        return _this;
    }
    AnatomogramContainerClass.prototype.setupDataServices = function () {
        this.registerDataset('cells', []);
        this.registerDataset('labels', []);
        reducer_1.BioblocksMiddlewareTransformer.addTransform(this.getCellToLabelTransform());
        reducer_1.BioblocksMiddlewareTransformer.addTransform(this.getLabelToCellTransform());
    };
    AnatomogramContainerClass.prototype.componentDidUpdate = function (prevProps) {
        var species = this.props.species;
        if (species !== prevProps.species) {
            this.setState({
                ids: this.deriveIdsFromSpecies(species),
            });
        }
    };
    AnatomogramContainerClass.prototype.render = function () {
        var _this = this;
        var _a = this.props, iconSrc = _a.iconSrc, species = _a.species, selectIds = _a.selectIds;
        var ids = this.state.ids;
        return (React.createElement(react_redux_1.Provider, { store: reducer_1.BBStore },
            React.createElement("div", { className: 'anatomogram-container', ref: function (node) {
                    if (node) {
                        _this.divRef = node.getElementsByTagName('div')[0];
                    }
                } },
                React.createElement(component_1.ComponentCard, { componentName: 'Anatomogram', iconSrc: iconSrc },
                    React.createElement(anatomogram_1.default, { atlasUrl: "", highlightColour: 'yellow', onClick: this.onClick, onInjected: this.resizeSVGElement, onMouseOut: this.onMouseOut, onMouseOver: this.onMouseOver, selectColour: 'ffaa00', selectIds: selectIds.toArray(), showIds: ids, species: species, selectedView: species === 'mus_musculus' ? 'female' : 'male' })))));
    };
    AnatomogramContainerClass.prototype.getCellToLabelTransform = function () {
        var _this = this;
        return {
            fn: function (state) {
                var e_1, _a, e_2, _b, e_3, _c;
                var species = _this.props.species;
                var currentCells = selector_1.selectCurrentItems(state, 'cells').toArray();
                var _d = selector_1.getSpring(state), category = _d.category, graphData = _d.graphData;
                var result = immutable_1.Set();
                try {
                    for (var currentCells_1 = tslib_1.__values(currentCells), currentCells_1_1 = currentCells_1.next(); !currentCells_1_1.done; currentCells_1_1 = currentCells_1.next()) {
                        var cellIndex = currentCells_1_1.value;
                        // Parse labels and categories from SPRING data.
                        var labelForCategory = graphData.nodes[cellIndex] ? graphData.nodes[cellIndex].labelForCategory : {};
                        var labels = data_1.AnatomogramMapping[species][labelForCategory[category]];
                        if (labels) {
                            labels.forEach(function (label) { return (result = result.add(label)); });
                        }
                        try {
                            for (var _e = (e_2 = void 0, tslib_1.__values(Object.keys(data_1.AnatomogramMapping[species]))), _f = _e.next(); !_f.done; _f = _e.next()) {
                                var id = _f.value;
                                try {
                                    for (var _g = (e_3 = void 0, tslib_1.__values(Object.values(labelForCategory))), _h = _g.next(); !_h.done; _h = _g.next()) {
                                        var label = _h.value;
                                        if (data_1.AnatomogramMapping[species][id].includes(label)) {
                                            result = result.add(id);
                                        }
                                    }
                                }
                                catch (e_3_1) { e_3 = { error: e_3_1 }; }
                                finally {
                                    try {
                                        if (_h && !_h.done && (_c = _g.return)) _c.call(_g);
                                    }
                                    finally { if (e_3) throw e_3.error; }
                                }
                            }
                        }
                        catch (e_2_1) { e_2 = { error: e_2_1 }; }
                        finally {
                            try {
                                if (_f && !_f.done && (_b = _e.return)) _b.call(_e);
                            }
                            finally { if (e_2) throw e_2.error; }
                        }
                    }
                }
                catch (e_1_1) { e_1 = { error: e_1_1 }; }
                finally {
                    try {
                        if (currentCells_1_1 && !currentCells_1_1.done && (_a = currentCells_1.return)) _a.call(currentCells_1);
                    }
                    finally { if (e_1) throw e_1.error; }
                }
                return result;
            },
            fromState: { stateName: 'cells' },
            toState: { stateName: 'labels' },
        };
    };
    AnatomogramContainerClass.prototype.getLabelToCellTransform = function () {
        var _this = this;
        return {
            fn: function (state) {
                var _a = _this.props, species = _a.species, selectIds = _a.selectIds;
                var graphData = selector_1.getSpring(state).graphData;
                var anatomogramMap = data_1.AnatomogramMapping[species];
                var candidateLabels = immutable_1.Set();
                selectIds.forEach(function (id) {
                    candidateLabels = id && anatomogramMap[id] ? candidateLabels.merge(anatomogramMap[id]) : candidateLabels;
                });
                var cellIndices = immutable_1.Set();
                graphData.nodes.forEach(function (node) {
                    var labelsForNode = Object.values(node.labelForCategory);
                    candidateLabels.forEach(function (label) {
                        if (label && labelsForNode.includes(label)) {
                            cellIndices = cellIndices.add(node.number);
                            return;
                        }
                    });
                });
                return cellIndices;
            },
            fromState: 'bioblocks/labels',
            toState: 'bioblocks/cells',
        };
    };
    AnatomogramContainerClass.defaultProps = {
        iconSrc: '',
        onLabelAdd: helper_1.EMPTY_FUNCTION,
        onLabelRemove: helper_1.EMPTY_FUNCTION,
        selectIds: immutable_1.Set(),
        species: 'homo_sapiens',
    };
    AnatomogramContainerClass.displayName = 'Anatomogram';
    return AnatomogramContainerClass;
}(container_1.BioblocksVisualization));
exports.AnatomogramContainerClass = AnatomogramContainerClass;
var mapStateToProps = function (state) { return ({
    selectIds: selector_1.selectCurrentItems(state, 'labels'),
}); };
var mapDispatchToProps = function (dispatch) {
    return redux_1.bindActionCreators({
        onLabelAdd: action_1.createContainerActions('labels').add,
        onLabelRemove: action_1.createContainerActions('labels').remove,
    }, dispatch);
};
exports.AnatomogramContainer = react_redux_1.connect(mapStateToProps, mapDispatchToProps)(AnatomogramContainerClass);
//# sourceMappingURL=AnatomogramContainer.js.map