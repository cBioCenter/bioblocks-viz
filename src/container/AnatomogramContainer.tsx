// tslint:disable-next-line:import-name
import Anatomogram from 'anatomogram';
import { isEqual } from 'lodash';
import * as React from 'react';

import {
  CellContext,
  ICellContext,
  initialCellContext,
  initialSpringContext,
  ISpringContext,
  SpringContext,
} from '~chell-viz~/context';
import { CHELL_CSS_STYLE } from '~chell-viz~/data';

export interface IAnatomogramContainerProps {
  cellContext: ICellContext;
  height: number | string;
  ids: string[];
  springContext: ISpringContext;
  style: CHELL_CSS_STYLE;
  width: number | string;
}

export interface IAnatomogramContainerState {
  selectIds: string[];
}

export const springToAnatomogramMapping: { [key: string]: string } = {
  P11A: 'UBERON_0000178',
  P11B: 'UBERON_0000473',
  P12A: 'UBERON_0000955',
  P9A: 'UBERON_0000977',
};

export const anatomogramToSpringMapping: { [key: string]: string } = {
  UBERON_0000178: 'P11A',
  UBERON_0000473: 'P11B',
  UBERON_0000955: 'P12A',
  UBERON_0000977: 'P9A',
};

export class AnatomogramContainerClass extends React.Component<IAnatomogramContainerProps, IAnatomogramContainerState> {
  public static defaultProps = {
    cellContext: {
      ...initialCellContext,
    },
    height: '300px',
    ids: ['UBERON_0000178', 'UBERON_0000473', 'UBERON_0000955', 'UBERON_0000977'],
    springContext: {
      ...initialSpringContext,
    },
    style: {},
    width: '400px',
  };

  public static displayName = 'Anatomogram';

  constructor(props: IAnatomogramContainerProps) {
    super(props);
    this.state = {
      selectIds: [],
    };
  }

  public componentDidUpdate(prevProps: IAnatomogramContainerProps) {
    const { cellContext, springContext } = this.props;
    if (!isEqual(springContext.selectedCategories, prevProps.springContext.selectedCategories)) {
      this.setState({
        selectIds: springContext.selectedCategories.map(category => springToAnatomogramMapping[category]),
      });
    } else if (!isEqual(cellContext.currentCells, prevProps.cellContext.currentCells)) {
      const categories = new Set<string>();
      for (const cellIndex of cellContext.currentCells) {
        categories.add(springContext.graphData.nodes[cellIndex].category);
      }
      this.setState({
        selectIds: Array.from(categories).map(category => springToAnatomogramMapping[category]),
      });
    }
  }

  public render() {
    const { selectIds } = this.state;

    return (
      <div className={'anatomogram-container'}>
        <Anatomogram
          atlasUrl={``}
          highlightColour={'yellow'}
          onClick={this.onClick}
          onMouseOut={this.onMouseOut}
          onMouseOver={this.onMouseOver}
          selectColour={'ffaa00'}
          selectIds={selectIds}
          showIds={this.props.ids}
          species={'homo_sapiens'}
        />
      </div>
    );
  }

  protected onClick = (id: string) => {
    this.props.springContext.toggleCategory(anatomogramToSpringMapping[id]);
  };

  protected onMouseOut = (id: string) => {
    return;
  };

  protected onMouseOver = (id: string) => {
    return;
  };
}

type requiredProps = Omit<IAnatomogramContainerProps, keyof typeof AnatomogramContainerClass.defaultProps> &
  Partial<IAnatomogramContainerProps>;

export const AnatomogramContainer = (props: requiredProps) => (
  <CellContext.Consumer>
    {cellContext => (
      <SpringContext.Consumer>
        {springContext => (
          <AnatomogramContainerClass {...props} cellContext={cellContext} springContext={springContext} />
        )}
      </SpringContext.Consumer>
    )}
  </CellContext.Consumer>
);
