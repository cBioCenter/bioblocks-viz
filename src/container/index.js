"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
tslib_1.__exportStar(require("./BioblocksVisualization"), exports);
tslib_1.__exportStar(require("./AnatomogramContainer"), exports);
tslib_1.__exportStar(require("./InfoPanelContainer"), exports);
tslib_1.__exportStar(require("./NGLContainer"), exports);
tslib_1.__exportStar(require("./ContactMapContainer"), exports);
tslib_1.__exportStar(require("./ProteinFeatureViewer"), exports);
tslib_1.__exportStar(require("./SpringContainer"), exports);
tslib_1.__exportStar(require("./TensorTContainer"), exports);
tslib_1.__exportStar(require("./UMAPSequenceContainer"), exports);
tslib_1.__exportStar(require("./UMAPTranscriptionalContainer"), exports);
//# sourceMappingURL=index.js.map