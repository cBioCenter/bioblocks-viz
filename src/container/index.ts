export * from './BioblocksVisualization';

export * from './AnatomogramContainer';
export * from './InfoPanelContainer';
export * from './NGLContainer';
export * from './ContactMapContainer';
export * from './ProteinFeatureViewer';
export * from './SpringContainer';
export * from './TensorTContainer';
export * from './UMAPSequenceContainer';
export * from './UMAPTranscriptionalContainer';
