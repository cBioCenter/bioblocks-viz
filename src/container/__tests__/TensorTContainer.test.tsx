import { shallow } from 'enzyme';
import * as React from 'react';

import { TensorTContainer } from '~chell-viz~/container';
import { genTensorTsneData } from '~chell-viz~/test';

describe('TensorTContainer', () => {
  it('Should match existing snapshot when given no props.', () => {
    const wrapper = shallow(<TensorTContainer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match existing snapshot when given sample data.', () => {
    const wrapper = shallow(<TensorTContainer data={genTensorTsneData()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
