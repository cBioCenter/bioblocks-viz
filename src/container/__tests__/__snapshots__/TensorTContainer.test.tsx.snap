// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TensorTContainer Should match existing snapshot when given no props. 1`] = `
<Component>
  <tSNE - TensorFlow
    cellContext={
      Object {
        "addCells": [Function],
        "currentCells": Array [],
        "removeAllCells": [Function],
        "removeCells": [Function],
      }
    }
    data={
      Array [
        Array [
          0,
        ],
        Array [
          0,
        ],
      ]
    }
    height={400}
    pointColor="#aa0000"
    springContext={
      Object {
        "addCategories": [Function],
        "addCategory": [Function],
        "graphData": Object {
          "links": Array [],
          "nodes": Array [],
        },
        "removeCategory": [Function],
        "selectedCategories": Array [],
        "setCategories": [Function],
        "toggleCategory": [Function],
      }
    }
    style={
      Object {
        "padding": 0,
      }
    }
    width={400}
  >
    <div
      id="TensorTContainerDiv"
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <Button
        as="button"
        compact={true}
        onClick={[Function]}
        role="button"
      >
        <button
          className="ui compact button"
          onClick={[Function]}
          role="button"
        >
          <Icon
            as="i"
            name="play"
          >
            <i
              aria-hidden="true"
              className="play icon"
            />
          </Icon>
          Play
        </button>
      </Button>
      <SettingsPanel
        configurations={
          Array [
            Object {
              "name": "Iterate Once",
              "onClick": [Function],
              "type": "BUTTON",
            },
            Object {
              "name": "Iterate Ten Times",
              "onClick": [Function],
              "type": "BUTTON",
            },
            Object {
              "name": "Iterate Fifty Times",
              "onClick": [Function],
              "type": "BUTTON",
            },
            Object {
              "name": "Reset",
              "onClick": [Function],
              "type": "BUTTON",
            },
            Object {
              "icon": "hashtag",
              "name": "Total Iterations: 0",
              "type": "LABEL",
            },
          ]
        }
        direction="left"
        inverted={true}
        opacity={0.8}
        showConfigurations={true}
        width="100%"
      >
        <div>
          <Grid
            columns={1}
          >
            <div
              className="ui one column grid"
            >
              <GridColumn>
                <div
                  className="column"
                >
                  <Button
                    as="button"
                    basic={true}
                    floated="right"
                    icon="settings"
                    onClick={[Function]}
                    role="button"
                  >
                    <button
                      className="ui basic icon right floated button"
                      onClick={[Function]}
                      role="button"
                    >
                      <Icon
                        as="i"
                        name="settings"
                      >
                        <i
                          aria-hidden="true"
                          className="settings icon"
                        />
                      </Icon>
                    </button>
                  </Button>
                </div>
              </GridColumn>
              <SidebarPushable
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="pushable"
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <Sidebar
                    animation="overlay"
                    as={[Function]}
                    direction="left"
                    duration={500}
                    inverted={true}
                    style={
                      Object {
                        "opacity": 0.8,
                        "width": "100%",
                      }
                    }
                    vertical={true}
                    visible={false}
                  >
                    <Ref
                      innerRef={[Function]}
                    >
                      <Menu
                        className="ui overlay left sidebar"
                        inverted={true}
                        style={
                          Object {
                            "opacity": 0.8,
                            "width": "100%",
                          }
                        }
                        vertical={true}
                      >
                        <div
                          className="ui inverted vertical ui overlay left sidebar menu"
                          style={
                            Object {
                              "opacity": 0.8,
                              "width": "100%",
                            }
                          }
                        >
                          <Grid
                            centered={true}
                          >
                            <div
                              className="ui centered grid"
                            >
                              <GridRow
                                key="iterate-once-0"
                              >
                                <div
                                  className="row"
                                >
                                  <Button
                                    as="button"
                                    compact={true}
                                    id="iterate-once-0"
                                    onClick={[Function]}
                                    role="button"
                                    style={Object {}}
                                  >
                                    <button
                                      className="ui compact button"
                                      id="iterate-once-0"
                                      onClick={[Function]}
                                      role="button"
                                      style={Object {}}
                                    >
                                      Iterate Once
                                    </button>
                                  </Button>
                                </div>
                              </GridRow>
                              <GridRow
                                key="iterate-ten-times-1"
                              >
                                <div
                                  className="row"
                                >
                                  <Button
                                    as="button"
                                    compact={true}
                                    id="iterate-ten-times-1"
                                    onClick={[Function]}
                                    role="button"
                                    style={Object {}}
                                  >
                                    <button
                                      className="ui compact button"
                                      id="iterate-ten-times-1"
                                      onClick={[Function]}
                                      role="button"
                                      style={Object {}}
                                    >
                                      Iterate Ten Times
                                    </button>
                                  </Button>
                                </div>
                              </GridRow>
                              <GridRow
                                key="iterate-fifty-times-2"
                              >
                                <div
                                  className="row"
                                >
                                  <Button
                                    as="button"
                                    compact={true}
                                    id="iterate-fifty-times-2"
                                    onClick={[Function]}
                                    role="button"
                                    style={Object {}}
                                  >
                                    <button
                                      className="ui compact button"
                                      id="iterate-fifty-times-2"
                                      onClick={[Function]}
                                      role="button"
                                      style={Object {}}
                                    >
                                      Iterate Fifty Times
                                    </button>
                                  </Button>
                                </div>
                              </GridRow>
                              <GridRow
                                key="reset-3"
                              >
                                <div
                                  className="row"
                                >
                                  <Button
                                    as="button"
                                    compact={true}
                                    id="reset-3"
                                    onClick={[Function]}
                                    role="button"
                                    style={Object {}}
                                  >
                                    <button
                                      className="ui compact button"
                                      id="reset-3"
                                      onClick={[Function]}
                                      role="button"
                                      style={Object {}}
                                    >
                                      Reset
                                    </button>
                                  </Button>
                                </div>
                              </GridRow>
                              <GridRow
                                key="total-iterations:-0-4"
                              >
                                <div
                                  className="row"
                                >
                                  <Label
                                    basic={true}
                                    color="orange"
                                    id="total-iterations:-0-4"
                                    style={Object {}}
                                  >
                                    <div
                                      className="ui orange basic label"
                                      id="total-iterations:-0-4"
                                      onClick={[Function]}
                                      style={Object {}}
                                    >
                                      Total Iterations: 0
                                    </div>
                                  </Label>
                                </div>
                              </GridRow>
                            </div>
                          </Grid>
                        </div>
                      </Menu>
                    </Ref>
                  </Sidebar>
                  <SidebarPusher>
                    <div
                      className="pusher"
                    >
                      <Component
                        height={400}
                        onSelectedCallback={[Function]}
                        pointsToPlot={Array []}
                        style={
                          Object {
                            "padding": 0,
                          }
                        }
                        width={400}
                      >
                        <TensorTComponentClass
                          height={400}
                          onSelectedCallback={[Function]}
                          pointsToPlot={Array []}
                          style={
                            Object {
                              "padding": 0,
                            }
                          }
                          width={400}
                        >
                          <div
                            id="TensorTComponentDiv"
                            style={
                              Object {
                                "padding": 0,
                              }
                            }
                          >
                            <PlotlyChart
                              config={Object {}}
                              data={Array []}
                              height="100%"
                              layout={
                                Object {
                                  "autosize": true,
                                  "dragmode": "select",
                                  "height": 400,
                                  "hovermode": "closest",
                                  "legend": Object {},
                                  "margin": Object {
                                    "b": 20,
                                  },
                                  "showlegend": false,
                                  "title": "",
                                  "width": 400,
                                  "xaxis": Object {
                                    "autorange": false,
                                    "range": Array [
                                      0,
                                      1,
                                    ],
                                    "showline": true,
                                  },
                                  "yaxis": Object {
                                    "autorange": false,
                                    "range": Array [
                                      0,
                                      1,
                                    ],
                                    "showline": true,
                                  },
                                }
                              }
                              onSelectedCallback={[Function]}
                              showLoader={true}
                              width="100%"
                            >
                              <div>
                                <Dimmer
                                  active={true}
                                >
                                  <DimmerInner
                                    active={true}
                                  >
                                    <div
                                      className="ui active transition visible dimmer"
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="content"
                                      >
                                        <Loader>
                                          <div
                                            className="ui loader"
                                          />
                                        </Loader>
                                      </div>
                                    </div>
                                  </DimmerInner>
                                </Dimmer>
                                <div
                                  className="plotly-chart"
                                  style={
                                    Object {
                                      "height": "100%",
                                      "marginBottom": 5,
                                      "width": "100%",
                                    }
                                  }
                                />
                              </div>
                            </PlotlyChart>
                          </div>
                        </TensorTComponentClass>
                      </Component>
                    </div>
                  </SidebarPusher>
                </div>
              </SidebarPushable>
            </div>
          </Grid>
        </div>
      </SettingsPanel>
    </div>
  </tSNE - TensorFlow>
</Component>
`;

exports[`TensorTContainer Should match existing snapshot when given sample data. 1`] = `
<Component
  data={
    Array [
      Array [
        -6.07794,
        0.66089,
        3.42628,
        1.85904,
        -2.41684,
        -0.43664,
        -2.30753,
        2.35615,
        -1.12598,
        -0.60199,
        -1.1218,
        0.11698,
        -0.04982,
        0.36092,
        -0.0508,
        -0.08095,
        0.80147,
        -0.4348,
        0.1589,
        -0.37422,
        -0.29722,
        0.25252,
        0.23296,
        -0.29226,
        -0.3324,
        -1.13309,
        0.37442,
        -0.0253,
        -0.02493,
        0.36059,
      ],
      Array [
        -5.00513,
        1.26166,
        4.98725,
        1.97746,
        -2.81049,
        -0.48861,
        0.39672,
        -0.04346,
        -1.18266,
        -0.22673,
        0.64698,
        -0.32172,
        -0.46212,
        -0.94495,
        -0.28995,
        -0.28883,
        -0.89687,
        -0.15866,
        -0.11644,
        1.05208,
        -0.51977,
        -0.30863,
        1.09945,
        0.41791,
        -0.56852,
        -0.2366,
        -0.1608,
        0.89874,
        -0.19515,
        0.05163,
      ],
    ]
  }
>
  <tSNE - TensorFlow
    cellContext={
      Object {
        "addCells": [Function],
        "currentCells": Array [],
        "removeAllCells": [Function],
        "removeCells": [Function],
      }
    }
    data={
      Array [
        Array [
          -6.07794,
          0.66089,
          3.42628,
          1.85904,
          -2.41684,
          -0.43664,
          -2.30753,
          2.35615,
          -1.12598,
          -0.60199,
          -1.1218,
          0.11698,
          -0.04982,
          0.36092,
          -0.0508,
          -0.08095,
          0.80147,
          -0.4348,
          0.1589,
          -0.37422,
          -0.29722,
          0.25252,
          0.23296,
          -0.29226,
          -0.3324,
          -1.13309,
          0.37442,
          -0.0253,
          -0.02493,
          0.36059,
        ],
        Array [
          -5.00513,
          1.26166,
          4.98725,
          1.97746,
          -2.81049,
          -0.48861,
          0.39672,
          -0.04346,
          -1.18266,
          -0.22673,
          0.64698,
          -0.32172,
          -0.46212,
          -0.94495,
          -0.28995,
          -0.28883,
          -0.89687,
          -0.15866,
          -0.11644,
          1.05208,
          -0.51977,
          -0.30863,
          1.09945,
          0.41791,
          -0.56852,
          -0.2366,
          -0.1608,
          0.89874,
          -0.19515,
          0.05163,
        ],
      ]
    }
    height={400}
    pointColor="#aa0000"
    springContext={
      Object {
        "addCategories": [Function],
        "addCategory": [Function],
        "graphData": Object {
          "links": Array [],
          "nodes": Array [],
        },
        "removeCategory": [Function],
        "selectedCategories": Array [],
        "setCategories": [Function],
        "toggleCategory": [Function],
      }
    }
    style={
      Object {
        "padding": 0,
      }
    }
    width={400}
  >
    <div
      id="TensorTContainerDiv"
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <Button
        as="button"
        compact={true}
        onClick={[Function]}
        role="button"
      >
        <button
          className="ui compact button"
          onClick={[Function]}
          role="button"
        >
          <Icon
            as="i"
            name="play"
          >
            <i
              aria-hidden="true"
              className="play icon"
            />
          </Icon>
          Play
        </button>
      </Button>
      <SettingsPanel
        configurations={
          Array [
            Object {
              "name": "Iterate Once",
              "onClick": [Function],
              "type": "BUTTON",
            },
            Object {
              "name": "Iterate Ten Times",
              "onClick": [Function],
              "type": "BUTTON",
            },
            Object {
              "name": "Iterate Fifty Times",
              "onClick": [Function],
              "type": "BUTTON",
            },
            Object {
              "name": "Reset",
              "onClick": [Function],
              "type": "BUTTON",
            },
            Object {
              "icon": "hashtag",
              "name": "Total Iterations: 0",
              "type": "LABEL",
            },
          ]
        }
        direction="left"
        inverted={true}
        opacity={0.8}
        showConfigurations={true}
        width="100%"
      >
        <div>
          <Grid
            columns={1}
          >
            <div
              className="ui one column grid"
            >
              <GridColumn>
                <div
                  className="column"
                >
                  <Button
                    as="button"
                    basic={true}
                    floated="right"
                    icon="settings"
                    onClick={[Function]}
                    role="button"
                  >
                    <button
                      className="ui basic icon right floated button"
                      onClick={[Function]}
                      role="button"
                    >
                      <Icon
                        as="i"
                        name="settings"
                      >
                        <i
                          aria-hidden="true"
                          className="settings icon"
                        />
                      </Icon>
                    </button>
                  </Button>
                </div>
              </GridColumn>
              <SidebarPushable
                style={
                  Object {
                    "width": "100%",
                  }
                }
              >
                <div
                  className="pushable"
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <Sidebar
                    animation="overlay"
                    as={[Function]}
                    direction="left"
                    duration={500}
                    inverted={true}
                    style={
                      Object {
                        "opacity": 0.8,
                        "width": "100%",
                      }
                    }
                    vertical={true}
                    visible={false}
                  >
                    <Ref
                      innerRef={[Function]}
                    >
                      <Menu
                        className="ui overlay left sidebar"
                        inverted={true}
                        style={
                          Object {
                            "opacity": 0.8,
                            "width": "100%",
                          }
                        }
                        vertical={true}
                      >
                        <div
                          className="ui inverted vertical ui overlay left sidebar menu"
                          style={
                            Object {
                              "opacity": 0.8,
                              "width": "100%",
                            }
                          }
                        >
                          <Grid
                            centered={true}
                          >
                            <div
                              className="ui centered grid"
                            >
                              <GridRow
                                key="iterate-once-0"
                              >
                                <div
                                  className="row"
                                >
                                  <Button
                                    as="button"
                                    compact={true}
                                    id="iterate-once-0"
                                    onClick={[Function]}
                                    role="button"
                                    style={Object {}}
                                  >
                                    <button
                                      className="ui compact button"
                                      id="iterate-once-0"
                                      onClick={[Function]}
                                      role="button"
                                      style={Object {}}
                                    >
                                      Iterate Once
                                    </button>
                                  </Button>
                                </div>
                              </GridRow>
                              <GridRow
                                key="iterate-ten-times-1"
                              >
                                <div
                                  className="row"
                                >
                                  <Button
                                    as="button"
                                    compact={true}
                                    id="iterate-ten-times-1"
                                    onClick={[Function]}
                                    role="button"
                                    style={Object {}}
                                  >
                                    <button
                                      className="ui compact button"
                                      id="iterate-ten-times-1"
                                      onClick={[Function]}
                                      role="button"
                                      style={Object {}}
                                    >
                                      Iterate Ten Times
                                    </button>
                                  </Button>
                                </div>
                              </GridRow>
                              <GridRow
                                key="iterate-fifty-times-2"
                              >
                                <div
                                  className="row"
                                >
                                  <Button
                                    as="button"
                                    compact={true}
                                    id="iterate-fifty-times-2"
                                    onClick={[Function]}
                                    role="button"
                                    style={Object {}}
                                  >
                                    <button
                                      className="ui compact button"
                                      id="iterate-fifty-times-2"
                                      onClick={[Function]}
                                      role="button"
                                      style={Object {}}
                                    >
                                      Iterate Fifty Times
                                    </button>
                                  </Button>
                                </div>
                              </GridRow>
                              <GridRow
                                key="reset-3"
                              >
                                <div
                                  className="row"
                                >
                                  <Button
                                    as="button"
                                    compact={true}
                                    id="reset-3"
                                    onClick={[Function]}
                                    role="button"
                                    style={Object {}}
                                  >
                                    <button
                                      className="ui compact button"
                                      id="reset-3"
                                      onClick={[Function]}
                                      role="button"
                                      style={Object {}}
                                    >
                                      Reset
                                    </button>
                                  </Button>
                                </div>
                              </GridRow>
                              <GridRow
                                key="total-iterations:-0-4"
                              >
                                <div
                                  className="row"
                                >
                                  <Label
                                    basic={true}
                                    color="orange"
                                    id="total-iterations:-0-4"
                                    style={Object {}}
                                  >
                                    <div
                                      className="ui orange basic label"
                                      id="total-iterations:-0-4"
                                      onClick={[Function]}
                                      style={Object {}}
                                    >
                                      Total Iterations: 0
                                    </div>
                                  </Label>
                                </div>
                              </GridRow>
                            </div>
                          </Grid>
                        </div>
                      </Menu>
                    </Ref>
                  </Sidebar>
                  <SidebarPusher>
                    <div
                      className="pusher"
                    >
                      <Component
                        height={400}
                        onSelectedCallback={[Function]}
                        pointsToPlot={Array []}
                        style={
                          Object {
                            "padding": 0,
                          }
                        }
                        width={400}
                      >
                        <TensorTComponentClass
                          height={400}
                          onSelectedCallback={[Function]}
                          pointsToPlot={Array []}
                          style={
                            Object {
                              "padding": 0,
                            }
                          }
                          width={400}
                        >
                          <div
                            id="TensorTComponentDiv"
                            style={
                              Object {
                                "padding": 0,
                              }
                            }
                          >
                            <PlotlyChart
                              config={Object {}}
                              data={Array []}
                              height="100%"
                              layout={
                                Object {
                                  "autosize": true,
                                  "dragmode": "select",
                                  "height": 400,
                                  "hovermode": "closest",
                                  "legend": Object {},
                                  "margin": Object {
                                    "b": 20,
                                  },
                                  "showlegend": false,
                                  "title": "",
                                  "width": 400,
                                  "xaxis": Object {
                                    "autorange": false,
                                    "range": Array [
                                      0,
                                      1,
                                    ],
                                    "showline": true,
                                  },
                                  "yaxis": Object {
                                    "autorange": false,
                                    "range": Array [
                                      0,
                                      1,
                                    ],
                                    "showline": true,
                                  },
                                }
                              }
                              onSelectedCallback={[Function]}
                              showLoader={true}
                              width="100%"
                            >
                              <div>
                                <Dimmer
                                  active={true}
                                >
                                  <DimmerInner
                                    active={true}
                                  >
                                    <div
                                      className="ui active transition visible dimmer"
                                      onClick={[Function]}
                                    >
                                      <div
                                        className="content"
                                      >
                                        <Loader>
                                          <div
                                            className="ui loader"
                                          />
                                        </Loader>
                                      </div>
                                    </div>
                                  </DimmerInner>
                                </Dimmer>
                                <div
                                  className="plotly-chart"
                                  style={
                                    Object {
                                      "height": "100%",
                                      "marginBottom": 5,
                                      "width": "100%",
                                    }
                                  }
                                />
                              </div>
                            </PlotlyChart>
                          </div>
                        </TensorTComponentClass>
                      </Component>
                    </div>
                  </SidebarPusher>
                </div>
              </SidebarPushable>
            </div>
          </Grid>
        </div>
      </SettingsPanel>
    </div>
  </tSNE - TensorFlow>
</Component>
`;
